packages/console/src/common/ConsoleUtils.ts

```diff
@@ -1,25 +1,45 @@
-import ShellQuote from 'shell-quote';
+import ShellQuote, { ParseEntry, ControlOperator } from 'shell-quote';
 import dh from '@deephaven/jsapi-shim';
 
 class ConsoleUtils {
+  static hasComment(arg: ParseEntry): arg is { comment: string } {
+    return (arg as { comment: string }).comment !== undefined;
+  }
+
+  static hasPattern(arg: ParseEntry): arg is { op: 'glob'; pattern: string } {
+    return (arg as { pattern: string }).pattern !== undefined;
+  }
+
+  static hasOp(arg: ParseEntry): arg is { op: ControlOperator } {
+    return (arg as { op: ControlOperator }).op !== undefined;
+  }
+
   /**
    * Given the provided text, parse out arguments using shell quoting rules.
    * @param str The text to parse.
    * @return string[] of the arguments. Empty if no arguments found.
    */
-  static parseArguments(str) {
+  static parseArguments(str: unknown): string[] {
     if (!str || !(typeof str === 'string' || str instanceof String)) {
```

Comment:
Not because of your change, but this seems weird to me - we return here if `str` is null or if `str` is not a string; but then below we do the filtering and check `hasComment`, `hasPattern`, etc. How is that possible?

Reply:
because in between, we are calling `ShellQuote.parse(str)`, thus turning `str` into `ParseEntry[]`. This ParseEntry type can be string, or an object with hasComment, etc.